src/main/java/org/opentripplanner/middleware/cdp/ConnectedDataManager.java

```diff
@@ -0,0 +1,216 @@
+    public static final String DATA_FILE_NAME_SUFFIX = FILE_NAME_SUFFIX + ".json";
+
+    private static final int CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES =
+        getConfigPropertyAsInt("CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES", 5);
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConnectedDataManager.class);
+
+    public static final String CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME =
+        getConfigPropertyAsText("CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME");
+
+    public static final String CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME =
+        getConfigPropertyAsText("CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME");
+
+    public static void scheduleTripHistoryUploadJob() {
+        LOG.info("Scheduling trip history upload for every {} minute(s)", CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES);
+        Scheduler.scheduleJob(
+            new TripHistoryUploadJob(),
+            0,
+            CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES,
+            TimeUnit.MINUTES);
+    }
+
+    /**
+     * Remove a user's trip requests and trip summaries from the database. Record the user's trip dates so that all
+     * data previously uploaded to s3 can be recompiled and uploaded again.
+     */
+    public static void removeUsersTripHistory(String userId) {
+        Set<Date> userTripDates = new HashSet<>();
+        for (TripRequest request : TripRequest.requestsForUser(userId)) {
+            userTripDates.add(getStartOfDay(request.dateCreated));
+            // This will delete the trip summaries as well.
+            request.delete();
+        }
+        // Get all dates that have already been earmarked for uploading.
+        Set<Date> incompleteUploads = new HashSet<>();
+        getIncompleteUploads().forEach(tripHistoryUpload -> incompleteUploads.add(tripHistoryUpload.uploadDate));
+        // Save all new dates for uploading.
+        Set<Date> newDates = Sets.difference(userTripDates, incompleteUploads);
+        TripHistoryUpload first = TripHistoryUpload.getFirst();
+        newDates.forEach(newDate -> {
+            if (first == null ||
+                newDate.getTime() == first.uploadDate.getTime() ||
+                (newDate.after(first.uploadDate) &&
+                newDate.before(getStartOfDay(new Date())))
+            ) {
+                // If the new date is the same or after the first ever upload date, add it to the upload list. This acts
+                // as a back stop to prevent historic uploads being created indefinitely. Also, make sure the new date
+                // is before today because the trip data for it isn't uploaded until after midnight.
+                Persistence.tripHistoryUploads.create(new TripHistoryUpload(newDate));
+            }
+        });
+        // Kick-off trip data processing to recompile and upload trip data to S3 minus the user's trip history.
+        processTripHistory();
```

Comment:
This could be a long-running process, so I think it'd be best to start a `TripHistoryUploadJob`.